### PR TITLE
RLintBear.py: Add version requirement

### PR DESCRIPTION
**Pull request by rohanraju97**
Closes https://github.com/coala/coala-bears/issues/1379

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: bears/r/RLintBear.py [hunk 1]
```diff
@@ -23,7 +23,8 @@ class RLintBear:
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     REQUIREMENTS = {RscriptRequirement(package='lintr', flag='-e',
-                                       repo='http://cran.rstudio.com')}
+                                       repo='http://cran.rstudio.com',
+                                       version='>=3.1.1')}
```
Comment (damngamerz) [inline bears/r/RLintBear.py]: Seems fine !! Have you tested it ?
Can you screen shot the test results here.

Comment (rohanraju97) [inline bears/r/RLintBear.py]: capture

Is this the one?

Comment (damngamerz) [inline bears/r/RLintBear.py]: @rohanraju97 hmm thats strange why all tests skipped ? this is out of my reach.let a maintainer reply to this.

Comment (Techievena) [inline bears/r/RLintBear.py]: This wont show up anything in bear-docs 
```

        if bear.REQUIREMENTS:
            output += "\n\nDependencies\n------------\n\n"
            outs = []
            for requirement in bear.REQUIREMENTS:
                if hasattr(requirement, 'type'):
                    outs.append("* ``" + requirement.type + "``"
                                " - ``" + requirement.package + "``")
                else:
                    outs.append("* System requirement\n")
                    for system, package in requirement.package.items():
             outs[-1] += "  - ``" + system + "`` - ``" + package + "``"
```
https://github.com/coala/bear-docs/blob/master/generate#L138

**Comment by rohanraju97** [comment]
@Techievena  So which file should I change exactly? I'm kinda lost here.

**Comment by Techievena** [comment]
@rohanraju97 You have to do changes here only ( I guess ). I think it's better if you do a bit of reading of generate file of bear-docs repo. You'll figure it out where to do the changes. I really can't do the reading right now. Got 2 tests tomorrow.

**Comment by rohanraju97** [comment]
Yeah. That's what I was going through. I'll try and figure it out. Thanks and all the best. :)

File: bears/r/RLintBear.py (hunk 1 above)
Comment (Techievena) [inline bears/r/RLintBear.py]: @damngamerz These tests always get skipped :smile: Moreover there is no need for tests here.

**Comment by rohanraju97** [comment]
@jayvdb  Can you help me out in figuring out this issue? I'm not sure if my change is correct or not.

**Comment by satwikkansal** [comment]
IMO just changing the `LANGUAGES` set is not enough to solve the issue (it's just the metadata)
Maybe you should use something like `check_prerequisites` to check whether needed runtime prerequisites of the bear are satisfied (the version of R in your case).

File: bears/r/RLintBear.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ class RLintBear:
     """
     Checks the code with ``lintr``.
     """
-    LANGUAGES = {'R'}
+    LANGUAGES = {'R >=3.1.1'}
```
Comment (satwikkansal) [inline bears/r/RLintBear.py]: You shouldn't add versions here.

**Comment by rohanraju97** [comment]
@satwikkansal Shouldn't the requirement be shown in the bear-docs repo? That is the issue.

**Comment by satwikkansal** [comment]
@rohanraju97 I don't think so otherwise it'd be marked as newcomer :P btw I think you should clarify your doubt by pinging the person who created the issue on the gitter channel. :)

File: bears/r/RLintBear.py (hunk 2 above)
Comment (rohanraju97) [inline bears/r/RLintBear.py]: Where do I do it then? I tried changing the Requirements dict in the source code of RLintBear in the last commit and I was told that that wasn't the right place either. I saw the generated bear docs repo and compared it with the source code and the place where I made the change now is the place that has all the metadata shown in the generated bear docs repo

Comment (jayvdb) [inline bears/r/RLintBear.py]: You could add a `DistributionRequirement` to the `REQUIREMENTS` below.

Comment (rohanraju97) [inline bears/r/RLintBear.py]: Would that change show in the bear-docs repo?

**Comment by jayvdb** [comment]
needs a rebase

**Comment by rohanraju97** [comment]
Done! @jayvdb

**Comment by sils** [comment]
ack https://github.com/coala/coala-bears/pull/1443/commits/b64a6da3a3744557052b8ea1ae416b96c5485ae1

**Comment by sils** [comment]
@rultor merge

**Pull request merged**